team/src/App.js

```diff
@@ -1,20 +1,81 @@
-import logo from './logo.svg';
+import React, { useState } from 'react';
+import { StreamChat } from 'stream-chat';
+import { Chat, enTranslations, Streami18n } from 'stream-chat-react';
+import { createGlobalStyle } from 'styled-components';
+import 'stream-chat-react/dist/css/index.css';
+
 import './App.css';
 
-function App() {
+import { ChannelContainer } from './components/ChannelContainer/ChannelContainer';
+import { ChannelListContainer } from './components/ChannelListContainer/ChannelListContainer';
+import { ColorSlider } from './components/ColorSlider/ColorSlider';
+
+const urlParams = new URLSearchParams(window.location.search);
+const apiKey = urlParams.get('apikey') || process.env.REACT_APP_STREAM_KEY;
+const user = urlParams.get('user') || process.env.REACT_APP_USER_ID;
+const theme = urlParams.get('theme') || 'light';
+const userToken =
+  urlParams.get('user_token') || process.env.REACT_APP_USER_TOKEN;
+
+const i18nInstance = new Streami18n({
+  language: 'en',
+  translationsForLanguage: {
+    ...enTranslations,
+    // '1 reply': '1 thread reply',
+    // '{{ replyCount }} replies': '{{ replyCount }} thread replies',
+    Flag: 'Pin to Channel',
+  },
+});
+
+const GlobalColor = createGlobalStyle`
+  body {
+    --primary-color: ${(props) => props.color};
+  }
+`;
+
+const filters = [{ type: 'team' }, { type: 'messaging' }];
+const options = { state: true, watch: true, presence: true, limit: 3 };
+const sort = { last_message_at: -1, updated_at: -1 };
+
+const App = () => {
+  const [createType, setCreateType] = useState('');
+  const [isCreating, setIsCreating] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [primaryColor, setPrimaryColor] = useState('78, 29, 157');
+
+  const client = new StreamChat(apiKey);
+  client.setUser({ id: user }, userToken);
+
   return (
-    <div className='App'>
-      <header className='App-header'>
-        <img src={logo} className='App-logo' alt='logo' />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a className='App-link' href='https://reactjs.org' target='_blank' rel='noopener noreferrer'>
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <GlobalColor color={primaryColor} />
+      <ColorSlider {...{ primaryColor, setPrimaryColor }} />
```

Comment:
This is problematic, it triggers queries on every change resulting in 100s of queries per second. Looks great, but it needs to not do a query at all. Can't we build this with just CSS and leave JS out of it by using css variables. 

@doodlemarks another problem is having the app in a container like this. As, at least initially we're going the iFrame route. This would bascially be the playing field: 
![image](https://user-images.githubusercontent.com/1172870/101885819-ebae4080-3b9a-11eb-9bba-374766d578a0.png)

This leaves us two options, one, put the slider inside the app (sidebar for example) or @DanC5 do the CSS variable thing and make sure we can trigger this with queryparams.

Reply:
@jaapbakker88 @tatianainama do either of you have a suggestion for how to pass a color from the slider into the app that doesn't result in so many renders? I'm not exactly sure how to do this exclusively with css variables since the color is set in JS. Currently we save the JS color and pass it into the css files via the styled components global style feature, but I'm not aware of how to get them to communicate using another method. Let me know if you have tips, thanks!

Reply:
Changed how this is done and now the app is not reloading on color seleciton, hopefully this works!